xapian-applications/omega/worker.cc

```diff
@@ -169,22 +171,19 @@ Worker::extract(const std::string& filename,
 		    !read_string(sockt, strpage)) {
 		    break;
 		}
-		if (strpage.empty() ||
-		    strpage.find_first_not_of("0123456789") != string::npos) {
+		if (!parse_signed(strpage.c_str(), pages) || pages < 0)
 		    pages = 0;
-		} else {
-		    pages = stoi(strpage);
-		}
 		return true;
 	    case MSG_NON_FATAL_ERROR:
-		if (strstate.length() > 1 && verbose) {
-		    cerr << (strstate.c_str() + 1) << endl;
+		if (strstate.length() > 1) {
+		    error = (strstate.c_str() + 1);
 		}
 		return false;
 	    default:
 		break;
 	}
     }
+    error = "The assistant process " + filter_module + " fails";
```

Comment:
The present tense sounds wrong for an error message - the event is in the past by the time the error message is read, so "X failed" sounds much more natural than "X fails".

Reply:
Okay, I will fix it.